r1cs/gadgets/std/src/fields/nonnative/nonnative_field_mul_result_gadget.rs

```diff
@@ -132,115 +145,61 @@ impl<SimulationF: PrimeField, ConstraintF: PrimeField>
-
-        debug_assert!(result.check(), "convert to mul result gagdet from non native gadget: check() failed on output gadget");
-
-        result
-    }
-
-    /// convert `self` to NonNativeFieldGadget. This functions assumes that `self` has the first
-    /// `params.num_limbs` set to 0, otherwise it is pointless to convert them to
-    /// a NonNativeFieldGadget.
-    // This function is currently only employed to convert back to NonNativeFieldGadget
-    // multiplication results for pseudo-mersenne field, for which such condition
-    // on the most significant limbs holds
-    fn to_non_native_field_gadget(&self, params: &NonNativeFieldParams) -> NonNativeFieldGadget<SimulationF, ConstraintF> {
-        debug_assert!(self.check(), "convert mul result gadget to non native gagdet: check failed on input gadget");
-
-        let limbs = self.limbs[params.num_limbs-1..].iter().cloned().collect();
-        // num_add of the NonNativeFieldGadget is 2^bits_per_limb times higher than the num_add of
-        // NonNativeFieldMulResultGadget, as the limbs in normal form are bits_per_limb smaller
-        let num_of_additions_over_normal_form = &self.num_add_over_normal_form*BigUint::from(2usize).pow(params.bits_per_limb as u32);
-        let result = NonNativeFieldGadget {
-            limbs,
-            num_of_additions_over_normal_form,
-            simulation_phantom: PhantomData,
-        };
-
-        debug_assert!(result.check(), "convert mul result gadget to non native gagdet: check failed on output gadget");
-
-        result
-    }
-
     /// Reducing a NonNativeMulResultGadget back to a non-native field gadget
-    /// in normal form. Assumes that 
+    /// in normal form for a non pseudo-mersenne field.
+    /// Assumes that
     /// ``
     ///     2 * bits_per_limb + surfeit' <= CAPACITY - 2.
     /// ``
-    /// where 
+    /// where
     /// ``
-    ///     bits_per_limb = NonNativeFieldParams::bits_per_limb, 
-    ///     surfeit' =  log(num_limbs + 2*(num_adds + 1)), 
+    ///     bits_per_limb = NonNativeFieldParams::bits_per_limb,
+    ///     surfeit' =  log(num_limbs + 2*(num_adds + 1)),
     ///     num_limbs = NonNativeFieldParams::num_limbs,
     /// ``
     /// and `num_adds` is as in the NonNativeMulResultGadget.
     // Costs
     // ``
-    //     C =  len(p) + surfeit' 
+    //     C =  len(p) + surfeit'
     //          + len(p) + num_limbs^2
     //          +  (num_groups - 1) * (3 + bits_per_limb + surfeit') + 1
     // ``
     // constraints, with `num_groups = Ceil(len(p) / S)` and
     // ``
     //  S - 1 = Floor[
     //          (ConstraintF::CAPACITY - 2 - (2 * bits_per_limb + surfeit') / bits_per_limb
-    //      ] 
+    //      ]
     //        = Floor[
     //          (ConstraintF::CAPACITY - 2 - surfeit') / bits_per_limb
     //      ] - 2.
     // ``
-    pub fn reduce<CS: ConstraintSystemAbstract<ConstraintF>>(
+    pub(crate) fn reduce_no_pseudomersenne<CS: ConstraintSystemAbstract<ConstraintF>>(
```

Comment:
As pointed out at the other comment of mine: Wouldn't it be better to call it `reduce_generic()`?

Reply:
Already done it besides renaming the function you mentioned before